### ISSUE TITLE
swarm: Confusing "This node is not a Manager" logs on Worker cluster join with token

### ISSUE THREAD

**Issue by abronan** [docker/docker#25014]
docker **master** with join-tokens

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Digital Ocean VMs:

_3 Managers_: 2 GB Memory / 40 GB Disk / Ubuntu 16.04 x64
_3 Agents_: 1 GB Memory / 30 GB Disk / Ubuntu 16.04 x64

**Steps to reproduce the issue:**
1. Init a Swarm
2. Register two additional managers with the manager token
3. Register 3 workers with the worker token

**Describe the results you received:**

I get confusing error logs on Worker joins, an example:

```
ERRO[0044] Error getting node 1y7jntucsh8fdk672ajv21zay: This node is not a swarm manager. Worker nodes can't be used to view or modify cluster state. Please run this command on a manager node or promote the current node to a manager. 
ERRO[0044] Handler for GET /v1.25/nodes/1y7jntucsh8fdk672ajv21zay returned error: This node is not a swarm manager. Worker nodes can't be used to view or modify cluster state. Please run this command on a manager node or promote the current node to a manager.
```

Although `docker node ls` gives a proper output and workers are effectively added:

```
root@node01:~# docker node ls
ID                           HOSTNAME  STATUS  AVAILABILITY  MANAGER STATUS
0qij1ldlpnowrt90xkguvefz1    node02    Ready   Active        Reachable
1y7jntucsh8fdk672ajv21zay    node04    Ready   Active        
910qp4r4t4c22ml9yo6p5cpl4 *  node01    Ready   Active        Leader
brhxqgmjgfqw6jpdkifb32ejk    node05    Ready   Active        
cu5tke0e7xwuz3427mmqtrrqi    node06    Ready   Active        
djtv63najxwmbjx0su281fdmo    node03    Ready   Active        Reachable
```

**Describe the results you expected:**

No error logs

**Additional information you deem important (e.g. issue happens only occasionally):**

It happens consistently so my guess is that there is some code that is dedicated to the Manager (node inspect) that is also run by the Worker when it shouldn't.

**Comment by aaronlehmann** [comment]
The join command figures out whether it has joined as a worker or manager by running a privileged command and seeing if it gets that error back. It's a bit of a hack. We'll probably have to improve it post-1.12.

**Comment by aaronlehmann** [comment]
This should be fixed by #27530.
